### PR TITLE
[unimodules] support static config for unimodule linking

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

Managed EAS requires ios and android projects which can be fully generated ([POC branch](https://github.com/expo/expo/compare/@evanbacon/ncl/auto-plugins?expand=1)). We could achieve this same effect with a dangerous mod, but bubbling up the static config seemed to make more sense (especially since we already kinda do with `excludedUnimodules`).

# How

Add support for a `react-native-unimodules` (explicitly named after the package) object which has `ios` and `android` objects that get forwarded to the autolinking scripts. The static config takes a lower priority than a manually defined config in the Podfile or gradle files.

# Test Plan

We use autolinking config in bare-expo, so switching to the new system and ensuring the projects build while still excluding packages seems ample.

### CODE REVIEW

File: packages/react-native-unimodules/README.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
\ No newline at end of file
+Learn more about this library and learn how to install it on the [Installing react-native-unimodules documentation page](https://docs.expo.io/bare/installing-unimodules/).
+
+## Config with package.json
+
+You can configure the iOS and Android linking using the `unimodules` object in the `package.json`:
+
+```json
+{
+  "unimodules": {
```
Comment (tsapeta) [inline packages/react-native-unimodules/README.md:11]: I wouldn't use `unimodules` anymore, especially in new places. We may want to deprecate `react-native-unimodules` and move this to the core or the adapter and call them "Expo modules".
Maybe we could already use `expo-modules` key? What do you think @ide?

Comment (ide) [reply packages/react-native-unimodules/README.md:11]: When we rename unimodules to Expo modules, we can rename the package.json key too.

Comment (EvanBacon) [reply packages/react-native-unimodules/README.md:11]: Agree with both. I'll add that this is a pretty niche feature that'll mostly be used to support monorepo managed workflow projects.

**Review by brentvatne — APPROVED** [review]
:shipit:

**Pull request merged**